x-pack/plugins/uptime/public/legacy_uptime/components/settings/settings_actions.tsx

```diff
@@ -41,21 +48,23 @@ export const SettingsActions = ({
         </WarningText>
       </EuiFlexItem>
       <EuiFlexItem grow={false}>
-        <EuiButtonEmpty
-          color="ghost"
-          size="s"
-          iconType="cross"
-          data-test-subj="discardSettingsButton"
-          isDisabled={!isFormDirty || isFormDisabled}
-          onClick={() => {
-            onCancel();
-          }}
-        >
-          <FormattedMessage
-            id="xpack.uptime.sourceConfiguration.discardSettingsButtonLabel"
-            defaultMessage="Cancel"
-          />
-        </EuiButtonEmpty>
+        <EuiThemeProvider colorMode="dark">
```

Comment:
why does it needs wrapping in EuiThemeProvider?

Reply:
Because the `color="ghost"` prop is being removed from EUI button components, and this is what replaces it. Internally, this is what the `color="ghost"` prop was doing in any case ([link to source code](https://github.com/elastic/eui/blob/1f8e4db288e01401d0eaeb11acb76a5f4318ec60/src/components/button/button.tsx#L113-L120)).

The deprecation is simply removing that extra piece of code complexity on EUI's end and asking that consumers who want always-dark colored buttons instead set a dark color mode wrapper themselves.

An extra point of nuance: if buttons are being used within an `<EuiBottomBar>`, the bottom bar component already sets its own dark theme provider wrapper on all children, so an extra wrapper isn't necessary. However, I can't tell if that's the case with this `legacy_uptime` button. It's being used within a `<SettingsBottomBar />` which I can't seem to determine is using a `<EuiBottomBar>` or not. If it's not and it's just an absolutely positioned bar with a black background, then the theme wrapper is still needed.